osu.Game/Overlays/DirectOverlay.cs

```diff
@@ -225,28 +246,139 @@ protected override void PopIn()
             base.PopIn();
 
             // Queries are allowed to be run only on the first pop-in
-            if (getSetsRequest == null)
+            if (beatmapSetPager == null)
                 queueUpdateSearch();
         }
 
-        private SearchBeatmapSetsRequest getSetsRequest;
-
         private readonly Bindable<string> currentQuery = new Bindable<string>(string.Empty);
 
         private ScheduledDelegate queryChangedDebounce;
+        private ScheduledDelegate addPageDebounce;
         private PreviewTrackManager previewTrackManager;
+        private BeatmapSetPager beatmapSetPager;
+
+        private class BeatmapSetPager
```

Comment:
Maybe a separate file for this? Blows up the size here kind of unnecessarily.

Reply:
Where should the file go?

Reply:
in it's own file, `BeatmapSetPager.cs`

Reply:
I was asking more about the directory, but I suppose you're implying that it should go in `osu.Game/Overlays`

Reply:
`osu.Game/Overlays/Direct` is probably best. If needed, can move up to `osu.Game/Online` somewhere.